src/bin/ipa_bench/gen_events.rs

```diff
@@ -16,35 +16,96 @@ use std::time::Duration;
 const RECORD_SEPARATOR: u8 = 30;
 
 const DAYS_IN_EPOCH: u64 = 7;
-type MatchKey = Vec<u64>;
+type MatchKey = u64;
 type Epoch = u8;
+type Number = u32;
 
 #[derive(Clone)]
 #[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
 pub struct EventBase {
-    // For this tool, we'll fix the length of a matchkey to u64
-    pub matchkeys: MatchKey,
+    /// Secret shared and then encrypted match keys.
+    pub matchkeys: Vec<MatchKey>,
```

Comment:
At the moment, we only have a solution for one match key. Shall we simplify it down to just one match key for now?

Reply:
Is one match key constraint permanent or temporary?
If we are going to support multiple matchkeys in the future, instead of changing the type, how about I change the data generation code to only generate one match key but in the vector. That way I can keep most of the current logics where I assume the data type is vector.

Reply:
OK, that's fine. Yes, eventually I would love to support multiple match keys.